public/EditorPage.tsx

```diff
@@ -0,0 +1,895 @@
+  try {
+    await navigator.clipboard.writeText(str);
+  } catch (err) {
+    console.error("Failed to copyStringToClipboard: ", err);
+  }
+};
+
+const EditorPageTopBar = () => {
+  const stateStore = useStateStore();
+  let {
+    faceConfig,
+    gallerySize,
+    setGallerySize,
+    gallerySectionConfigList,
+    shuffleGenderSettingObject,
+    shuffleRaceSettingObject,
+    setShuffleGenderSettingObject,
+    setShuffleRaceSettingObject,
+  } = stateStore;
+  const navigate = useNavigate();
+
+  const genders = ["male", "female"];
+  const races = ["white", "black", "brown", "asian"];
+
+  const [genderInvalidOptions, setGenderInvalidOptions] =
+    useState<boolean>(false);
+  const [raceInvalidOptions, setRaceInvalidOptions] = useState<boolean>(false);
+
+  const optionsNotDisplayed =
+    genders.length +
+    races.length -
+    (shuffleGenderSettingObject.length + shuffleRaceSettingObject.length);
+
+  return (
+    <div className="bg-slate-800 text-white flex justify-between w-full fixed z-50	">
+      <div className="flex gap-4 text-xl p-2 justify-around items-center">
+        <span className="cursor-pointer rounded-full p-1 m-0.5 hover:bg-slate-50 hover:text-slate-900">
+          <House weight="fill" size={24} onClick={() => navigate("/")} />
+        </span>
+        <span className="hidden md:inline">faces.js Editor</span>
+        <Badge
+          content={optionsNotDisplayed ? optionsNotDisplayed : null}
+          shape="rectangle"
+          isInvisible={!optionsNotDisplayed}
+        >
+          <ButtonGroup>
+            <Button
+              className="bg-slate-800 text-white border-2 border-white"
+              onClick={() =>
+                shuffleEntireFace(
+                  faceConfig,
+                  gallerySectionConfigList,
+                  stateStore,
+                )
+              }
+            >
+              <Shuffle size={24} />
+            </Button>
+            <Popover placement="bottom" showArrow offset={10}>
+              <PopoverTrigger>
+                <Button className="bg-slate-800 text-white border-t-2 border-r-2 border-b-2 border-white">
+                  <Sliders size={24} />
+                </Button>
+              </PopoverTrigger>
+              <PopoverContent className="">
+                {(titleProps) => (
+                  <div className="px-1 py-2 w-full">
+                    <p
+                      className="text-small font-bold text-foreground"
+                      {...titleProps}
+                    >
+                      Customize Shuffle
+                    </p>
+                    <div className="flex gap-4">
+                      <CheckboxGroup
+                        label="Gender"
+                        isInvalid={genderInvalidOptions}
+                        value={shuffleGenderSettingObject}
+                        errorMessage={
+                          genderInvalidOptions ? "Select at least one" : null
```

Comment:
If none is selected, it still lets you shuffle, seemingly with the default value (male and all races) - maybe would be better to say "Will use default if none are selected" or something.